Replace xerrors

sespiros opened the pull request:
As of Go 1.13, Errorf's %w verb has been incorporated into fmt.Errorf and the As and Is functions have been incorporated into the standard library's errors package.

https://pkg.go.dev/golang.org/x/xerrors#pkg-overview

Commit:
multiple: replace xerrors.Errorf instances with fmt.Errorf

as of Go 1.13, Errorf's %w verb has been incorporated into fmt.Errorf.

Commit:
multiple: replace xerrors.Is and As with errors.Is and As

as of Go 1.13, Is, As and Unwrap have been incorporated into the
standard library's errors package.

Commit:
linux/filepath.go: replace xerrors.Errorf with fmt.Errorf and use %d for uint64

Commit:
go.mod, go.sum: ran go mod tidy

chrisccoulson reviewed (changes requested):
Thanks - it looks like the error strings in the tests in `internal/ioerr` need updating. Note that the `xerrors` package had a bug where putting the `%w` verb anywhere other than at the end of a format string would result in the error being printed twice, and the tests accommodated this, but it looks like the `fmt` package doesn't have this same bug.

Commit:
i/ioerr/ioerr_test.go: fix tests

xerrors package had a bug where putting the %w verb anywhere other
than at the end of a format string would result in the error being
printed twice, and the tests accomodated this, but it looks like the
fmt package doesn't have the same bug. (from Chris)

sespiros commented:
> Thanks - it looks like the error strings in the tests in `internal/ioerr` need updating. Note that the `xerrors` package had a bug where putting the `%w` verb anywhere other than at the end of a format string would result in the error being printed twice, and the tests accommodated this, but it looks like the `fmt` package doesn't have this same bug.

Thanks! Fixed the tests and rebased (I forgot to run the tests with `go test ./...` so I missed those).

chrisccoulson reviewed (approved):
Thanks, that looks good to me
